src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> next;
+        private Node<T> prev;
+        private T element;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.next = next;
+            this.prev = prev;
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> newNode;
+        if (tail == null) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            validateIndex(index);
+            Node<T> nodeOnIndex = findNodeOnIndex(index);
+            Node<T> prevNodeOfFound = nodeOnIndex.prev;
+            Node<T> nodeToAdd = new Node<>(prevNodeOfFound, value, nodeOnIndex);
+            nodeOnIndex.prev = nodeToAdd;
+            if (prevNodeOfFound == null) {
+                head = nodeToAdd;
+            } else {
+                prevNodeOfFound.next = nodeToAdd;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> currentNode = findNodeOnIndex(index);
+        return currentNode.element;
```

Comment:
```suggestion
        return findNodeOnIndex(index).element;
```
no need to create a redundant variable